src/app/category.service.ts

```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CategoryService {
```

Comment:
add a category interface like you did with products. If you dont the value of using typescript is kind of lost